include/mockturtle/generators/arithmetic.hpp

```diff
@@ -183,4 +226,85 @@ inline std::vector<signal<Ntk>> carry_ripple_multiplier( Ntk& ntk, std::vector<s
   return res;
 }
 
+/*! \brief Creates a sideways sum adder using half and full adders
+ *
+ * The function creates the adder in `ntk` and returns output signals,
+ * whose size is floor(log2(a.size()))+1.
+ *
+ * \param ntk Network
+ * \param a Input operand
+ */
+template<typename Ntk>
+inline std::vector<signal<Ntk>> sideways_sum_adder( Ntk& ntk, std::vector<signal<Ntk>> const& a )
+{
+  static_assert( is_network_type_v<Ntk>, "Ntk is not a network type" );
+
+  int n = a.size();
+
+  int out_n = 1; // floor(log2(n) + 1)
+  int tmpn = n;
+  while( tmpn >>= 1 ) out_n++;
+
+  std::list<signal<Ntk>> *first_level  = new std::list<signal<Ntk>>; // sum bits
+  std::list<signal<Ntk>> *second_level = new std::list<signal<Ntk>>; // carry bits
+
+  auto res = constant_word( ntk, 0, out_n );
+  int output_ind = 0;
+
+  for(int i = 0; i < n; i++)
+	first_level->push_back(a[i]);
+
+  while(1)
+  {
+    while(!first_level->empty())
+	{
+      if(first_level->size() == 1)
+	  {
+        auto in_sig = first_level->front();
+        first_level->pop_front();
+        res[output_ind++] = in_sig;
+      }
+      else if(first_level->size() == 2)
+      {
+        auto in_sig1 = first_level->front();
+        first_level->pop_front();
+        auto in_sig2 = first_level->front();
+        first_level->pop_front();
+        signal<Ntk> tmp_sum;
+		signal<Ntk> tmp_carry;
+        std::tie( tmp_sum, tmp_carry ) = half_adder( ntk, in_sig1, in_sig2 );
+		first_level->push_back(tmp_sum);
+		second_level->push_back(tmp_carry);
+      }
+      else // first_level->size() >=3
+      {
+	    auto in_sig1 = first_level->front();
+        first_level->pop_front();
+        auto in_sig2 = first_level->front();
+        first_level->pop_front();
+		auto in_sig3 = first_level->front();
```

Comment:
There are some tabs in the code, which should be spaces.

Reply:
Vim does that automaticaly for me, will fix that.